sql/information_schema/columns_table.go

```diff
@@ -185,45 +186,41 @@ func columnsRowIter(ctx *sql.Context, catalog sql.Catalog, allColsWithDefaultVal
 // getRowFromColumn returns a single row for given column. The arguments passed are used to define all row values.
 // These include the current ordinal position, so this column will get the next position number, sql.Column object,
 // database name, table name, column key and column privileges information through privileges set for the table.
-func getRowFromColumn(ctx *sql.Context, curOrdPos int, col *sql.Column, dbName, tblName, columnKey string, privSetTbl sql.PrivilegeSetTable, privSetMap map[string]struct{}) sql.Row {
+func getRowFromColumn(ctx *sql.Context, curOrdPos int, col *sql.Column, dbCollation sql.CollationID, dbName, tblName, columnKey string, privSetTbl sql.PrivilegeSetTable, privSetMap map[string]struct{}) sql.Row {
 	var (
 		charName          interface{}
 		collName          interface{}
 		charMaxLen        interface{}
 		charOctetLen      interface{}
 		nullable          = "NO"
 		ordinalPos        = uint32(curOrdPos + 1)
-		colType           = strings.Split(col.Type.String(), " COLLATE")[0]
-		dataType          = colType
 		datetimePrecision interface{}
 		srsId             interface{}
 	)
 
+	colType, dataType := getDtdIdAndDataType(col.Type)
+
 	if col.Nullable {
 		nullable = "YES"
 	}
 
 	if types.IsText(col.Type) {
 		if types.IsTextOnly(col.Type) {
-			charName = sql.Collation_Default.CharacterSet().String()
-			collName = sql.Collation_Default.String()
+			charName = dbCollation.CharacterSet().String()
```

Comment:
I don't think we need to use the database collation here. The collation will be in the column's type itself, which you can get via the `sql.TypeWithCollation` interface (that `Text`, `Set`, and `Enum` types implement).